model/build/build.go

```diff
@@ -63,6 +63,37 @@ func (b *Build) IsFinished() bool {
 		b.Status == evergreen.BuildSucceeded
 }
 
+func taskCacheTaskIsUnscheduled(t *TaskCache) bool {
+	if !t.Activated && t.Status == evergreen.TaskUndispatched {
+		return true
+	}
+
+	return false
+}
+
+// AllCachedTasksOrCompileFinished returns true when either:
+//  1. if there is a compile task, the compile task's status is one the ones
+//     listed in IsFinishedTaskStatus
```

Comment:
`IsFailedTaskStatus` (I started writing a comment explaining why this is wrong, then realized that the function implements this correctly, but the comment is incorrect.)

Reply:
fixed